core/src/test/java/org/elasticsearch/cluster/metadata/WildcardExpressionResolverTests.java

```diff
@@ -125,6 +126,47 @@ public void testAll() {
         assertThat(newHashSet(resolver.resolve(context, Arrays.asList("_all"))), equalTo(newHashSet("testXXX", "testXYY", "testYYY")));
     }
 
+    // issue #23960
```

Comment:
I think a lot of what you're looking for is not written down, it's tribal knowledge that we all accumulate with experience in the project. We explain some straightforward things in the contributing docs but the vast majority of our conventions are not written down (and I think that's okay, we catch them during code review).